app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -95,6 +92,8 @@ def fetch_domain_servers(feed)
           server = persister.middleware_servers.find_or_build(domain_server.path)
           parse_middleware_server(domain_server, server, true, server_name)
 
+          associate_with_rhevm(server, feed)
```

Comment:
Per each feed, we list the domain servers and then for this `<server, feed>` pair we try add the association by running this method. The `associate_with_rhevm` figures out the `machine_id` (it was stored under the operating system resource in the inventory, if I recall correctly) and based on this id the linking is done. This will work for domain mode that has only one host - the default scenario. For multiple hosts, it will show that the servers are running on the domain controller host, but this wont be the true. Unfortunately, we don't have this info (machine id per host controller) in the inventory, so this can be improved in the future.